interfaces/cython/cantera/ctml2yaml.py

```diff
@@ -0,0 +1,1137 @@
+            if falloff_type != "Troe":
+                raise TypeError(
+                    "Unknown activation type '{}' for reaction id {}".format(
+                        falloff_type, reaction.get("id")
+                    )
+                )
+
+        func = getattr(self, reaction_type.lower())
+        reaction_attribs.update(func(rate_coeff))
+
+        reaction_attribs["equation"] = (
+            # This has to replace the reaction direction symbols separately because
+            # species names can have [ or ] in them
+            reaction.find("equation")
+            .text.replace("[=]", "<=>")
+            .replace("=]", "=>")
+        )
+        negative_A = reaction.get("negative_A")
+        if negative_A is not None and negative_A == "yes":
+            reaction_attribs["negative-A"] = True
+
+        reactants = split_species_value_string(reaction.findtext("reactants"))
+        # products = {
+        #     a.split(":")[0]: float(a.split(":")[1])
+        #     for a in reaction.findtext("products").replace("\n", " ").strip().split()
+        # }
+        orders = {}
+        # Need to make this more general, for non-reactant orders
+        for order_node in reaction.iterfind("order"):
+            species = order_node.get("species")
+            order = float(order_node.text)
+            if not np.isclose(reactants[species], order):
+                orders[species] = order
+        if orders:
+            reaction_attribs["orders"] = orders
+
+        if reaction.get("negative_orders") == "yes":
+            reaction_attribs["negative-orders"] = True
+
+        if reaction.get("duplicate", "") == "yes":
+            reaction_attribs["duplicate"] = True
+
+        self.reaction_attribs = reaction_attribs
+
+    @classmethod
+    def to_yaml(cls, representer, data):
+        return representer.represent_dict(data.reaction_attribs)
+
+    def threebody(self, rate_coeff):
+        """Process a three-body reaction.
+
+        Returns a dictionary with the appropriate fields set that is
+        used to update the parent reaction entry dictionary.
+        """
+        reaction_attribs = FlowMap({"type": "three-body"})
+        reaction_attribs["rate-constant"] = self.process_arrhenius_parameters(
+            rate_coeff.find("Arrhenius")
+        )
+        eff_node = rate_coeff.find("efficiencies")
+        if eff_node is not None:
+            reaction_attribs["efficiencies"] = self.process_efficiencies(eff_node)
+
+        return reaction_attribs
+
+    def lindemann(self, rate_coeff):
+        """Process a Lindemann falloff reaction.
+
+        Returns a dictionary with the appropriate fields set that is
+        used to update the parent reaction entry dictionary.
+        """
+        reaction_attribs = FlowMap({"type": "falloff"})
+        for arr_coeff in rate_coeff.iterfind("Arrhenius"):
+            if arr_coeff.get("name") is not None and arr_coeff.get("name") == "k0":
```

Comment:
Changed this case (and others) to:
```python
if xxx.get("var", "").lower() == "value":
    ...
```
since I don't think case-sensitivity is enforced for any of the XML attribute values.

Reply:
The default value of `""` is necessary since if the value doesn't exist, `.lower()` is not an attribute of `None`.